tests/DocumentationExamplesTest.php

```diff
@@ -1022,6 +1029,128 @@ public function testChangeStreamExample_1_4()
         $this->assertNull($nextChange);
     }
 
+    public function testAggregation_example_1()
+    {
+        $db = new Database($this->manager, $this->getDatabaseName());
+        $cursor = $db->sales->aggregate([
+            ['$match' => ['items.fruit' => 'banana']],
+            ['$sort' => ['date' => 1]]
+        ]);
+        $this->assertInstanceOf('MongoDB\Driver\Cursor', $cursor);
+    }
+
+    public function testAggregation_example_2()
+    {
+        $db = new Database($this->manager, $this->getDatabaseName());
+        $cursor = $db->sales->aggregate([
+            ['$unwind' => '$items'],
+            ['$match' => ['items.fruit' => 'banana']],
+            [
+                '$group' => ['_id' => ['day' => ['$dayOfWeek' => '$date']],
+                'count' => ['$sum' => '$items.quantity']]
+            ],
+            [
+                '$project' => [
+                    'dayOfWeek' => '$_id.day',
+                    'numberSold' => '$count',
+                    '_id' => 0
+                ]
+            ],
+            ['$sort' => ['numberSold' => 1]]
+        ]);
+        $this->assertInstanceOf('MongoDB\Driver\Cursor', $cursor);
+    }
+
+    public function testAggregation_example_3()
+    {
+        $db = new Database($this->manager, $this->getDatabaseName());
+        $cursor = $db->sales->aggregate([
+            ['$unwind' => '$items'],
+            ['$group' => [
+                '_id' => ['day' => ['$dayOfWeek' => '$date']],
+                'items_sold' => ['$sum' => '$items.quantity'],
+                'revenue' => [
+                    '$sum' => [
+                        '$multiply' => ['$items.quantity', '$items.price']
+                    ]
+                ]
+            ]],
+            ['$project' => [
+                'day' => '$_id.day', 'revenue' => 1, 'items_sold' => 1,
+                'discount' => [
+                    '$cond' => [
+                        'if' => ['$lte' => ['$revenue', 250]],
+                        'then' => 25,
+                        'else' => 0
+                    ]
+                ]
+            ]]
+        ]);
+        $this->assertInstanceOf('MongoDB\Driver\Cursor', $cursor);
+    }
+
+    public function testAggregation_example_4()
+    {
+        if (version_compare($this->getServerVersion(), '3.6.0', '<')) {
+            $this->markTestSkipped('let of an object is not supported');
```

Comment:
Is there a citation for this? I didn't see anything mentioned in the [3.6 release notes](https://docs.mongodb.com/manual/release-notes/3.6/) and [`$let` documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/let/) doesn't talk about version differences.

Reply:
This test fails on 3.2 and 3.4 with `arguments to $lookup must be strings, let: { constituents: "$airlines" } is type object`. [This](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/index.html#join-conditions-and-uncorrelated-sub-queries) is the only documentation I could find that might explain why.